### PR TITLE
Test of pipeline-library#583

### PR DESCRIPTION
**Pull request by lemeurherve**
Test of https://github.com/jenkins-infra/pipeline-library/pull/583

### CODE REVIEW

**Comment by lemeurherve** [comment]
<details>
<summary>Without skip label: ✅ </summary>

image

image

</details>

<details>
<summary>With skip label: ✅ </summary>

image

image

</details>

**Comment by lemeurherve** [comment]
Closing as successfully tested.

**Comment by lemeurherve** [comment]
Reopened to test the last commits (flattened logic and try/catch around the github app credential use), up to https://github.com/jenkins-infra/pipeline-library/pull/583/commits/1a2c415432301df5fb9d01e1652ab349dd0004be

**Comment by timja** [comment]
> Reopened to test the last commits (flattened logic and try/catch around the github app credential use), up to [jenkins-infra/pipeline-library@1a2c415](https://github.com/jenkins-infra/pipeline-library/commit/1a2c415432301df5fb9d01e1652ab349dd0004be)

you can lookup the credential the build is running with and not hardcode it with this:
https://github.com/hmcts/cnp-jenkins-library/blob/f47e524c7b6d4f3e77e8a2d933cf795a94173aba/vars/checkoutScm.groovy#L24-L25

**Comment by lemeurherve** [comment]
> > Reopened to test the last commits (flattened logic and try/catch around the github app credential use), up to [jenkins-infra/pipeline-library@1a2c415](https://github.com/jenkins-infra/pipeline-library/commit/1a2c415432301df5fb9d01e1652ab349dd0004be)
> 
> you can lookup the credential the build is running with and not hardcode it with this: [hmcts/cnp-jenkins-library@`f47e524`/vars/checkoutScm.groovy#L24-L25](https://github.com/hmcts/cnp-jenkins-library/blob/f47e524c7b6d4f3e77e8a2d933cf795a94173aba/vars/checkoutScm.groovy#L24-L25)

Perfect, many thanks @timja!

File: Jenkinsfile [hunk 1]
```diff
@@ -2,4 +2,5 @@
  See the documentation for more options:
  https://github.com/jenkins-infra/pipeline-library/
 */
+@Library('pipeline-library@pull/583/head') _
```
Comment (lemeurherve) [inline Jenkinsfile:5]:
```suggestion
@Library('pipeline-library@pull/584/head') _
```

**Comment by lemeurherve** [comment]
Test of PR 584 successful 🎉
 
https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fjenkins-infra-test-plugin/detail/PR-64/26/pipeline/101

The Linux build skipped the ACP as it ran when the label was applied, and the Windows one used the Azure ACP as I removed the label in the meantime.

**Comment by lemeurherve** [comment]
Test of 583 with 584 included OK too: https://ci.jenkins.io/job/Plugins/job/jenkins-infra-test-plugin/job/PR-64/27/console 🚀

**Comment by lemeurherve** [comment]
pipeline-lib PR merged 🙂

**Pull request closed without merging**